compiler/annotation/pipeline.rs

```diff
@@ -290,6 +290,18 @@ fn annotate_stage(
                         insert_annotations.vertex_annotations_of(role_name.type_()).unwrap().clone(),
                     );
                 }
+                Constraint::Links(links) => {
+                    if let Some(variable) = links.role_type().as_variable() {
+                        if !running_variable_annotations.contains_key(&variable)
+                            && insert_annotations.vertex_annotations_of(links.role_type()).is_some()
+                        {
+                            running_variable_annotations.insert(
+                                variable,
+                                insert_annotations.vertex_annotations_of(links.role_type()).unwrap().clone(),
+                            );
+                        }
+                    }
+                }
```

Comment:
Vertices corresponding to the role may also be freshly added by the insert clause.